kv/txn_coord_sender_test.go

```diff
@@ -727,6 +727,41 @@ func TestTxnCoordSenderErrorWithIntent(t *testing.T) {
 	}
 }
 
+// TestTxnCoordSenderInternalError is similar to TestTxnCoordSenderErrorWithIntent,
+// but the send returns an internal error instead of NewTransactionRetryError.
+func TestTxnCoordSenderInternalError(t *testing.T) {
```

Comment:
this is a lot of copy pasta, would it be able to merge those two tests? No need for an elaborate setup, just return an error based on transaction name and check that you get two tracked txns in the end.